managers/views.py

```diff
@@ -38,10 +40,67 @@ def get_queryset(self):
         return projects_queryset
 
 
+class ProjectCreate(APIView):
+    renderer_classes = [renderers.TemplateHTMLRenderer]
+    template_name = 'managers/project_create.html'
+
+    def get(self, request):
+        project_serializer = ProjectCreateSerializer(context={'request': request})
+        return Response({'serializer': project_serializer})
+
+    def post(self, request):
+        project_serializer = ProjectCreateSerializer(data=request.data, context={'request': request})
+        if not project_serializer.is_valid():
+            return Response({
+                'serializer': project_serializer,
+                'errors': project_serializer.errors,
+            })
+        project_serializer.save()
+        return redirect('custom-projects-list')
+
+
 class ProjectDetail(APIView):
     renderer_classes = [renderers.TemplateHTMLRenderer]
     template_name = 'managers/project_detail.html'
 
     def get(self, request, pk):
         project = get_object_or_404(Project, pk=pk)
         return Response({'project': project})
+
+
+class ProjectUpdate(APIView):
+    renderer_classes = [renderers.TemplateHTMLRenderer]
+    template_name = 'managers/project_update.html'
+
+    def get(self, request, pk):
+        project = get_object_or_404(Project, pk=pk)
+        project_serializer = ProjectSerializer(
+            project,
+            context={'request': request},
+        )
+        return Response({'serializer': project_serializer, 'project': project})
+
+    def post(self, request, pk):
+        project = get_object_or_404(Project, pk=pk)
+        project_serializer = ProjectSerializer(
+            project,
+            data=request.data,
+            context={'request': request},
+        )
+        if not project_serializer.is_valid():
+            return Response({
+                'serializer': project_serializer,
+                'project': project,
+                'errors': project_serializer.errors,
+            })
+        project_serializer.save()
+        return redirect('custom-project-detail', pk=pk)
+
+
+def delete_project(request, pk):
+    if request.user.user_type == CustomUser.UserType.ADMIN.name:
```

Comment:
@Karrp @MartynaAnnaGottschling Question about how this should exactly work - here implementation shows that only `ADMIN` should be allowed to perform this action. In template, everyone can click on this (which is a little bit miselading, I think it should be shown only for admins), however the real issue is that admins doesn't have access Projects List from the menu.

Reply:
Hey, might be late to respond but unless you already applied some fixes, I can navigate to project list from menu, while signed in as admin.

And since all delete operations (except for individual reports) should be performed only by admin, you are right that the delete button should be visible only to admin.

Reply:
But the delete button is in update template, which is accessable from detail template, only for admin, I think...

Reply:
@MartynaAnnaGottschling @maciejSamerdak I think I was able to reach it as manager too. Anyway, the point is that if it should be doable only by admin, only admin should be able to click on it - it might be bad user experience if there will be a button visible for users that when clicked will redirect them to `home` without telling them what actually happened. So we should agree on who can do and see what - for me both ways make sense (admin and manager who created this project or just admin), but whatever we choose lets be consistent.

We will probably solve more of the permissions cases in the future so lets just focus on this one now.

Reply:
@Karrp says only admin should be able to delete a project.